extensions/barrage/src/main/java/io/deephaven/extensions/barrage/util/BarrageUtil.java

```diff
@@ -452,6 +452,8 @@ private static ConvertedArrowSchema convertArrowSchema(
                 // TODO (deephaven-core#3403) widen this check for better assurances
                 Class<?> defaultType = getDefaultType(getArrowType.apply(i), result, i);
                 Assert.eq(type.getValue(), "deephaven column type", defaultType, "arrow inferred type");
```

Comment:
Colin and I would prefer to, eventually, create more assurances that the metadata types match the incoming arrow data type. Otherwise, there are either silent failures with bad parsed data (when the types are the same size) or we throw exceptions when parsing with the mismatches. So, let's keep the check.

I do agree we should change `getDefaultType` to return `Boolean.class` instead of `boolean.class`.